packages/visual-editor/src/components/Dropzone/Hitboxes.tsx

```diff
@@ -0,0 +1,206 @@
+  return INITIAL_OFFSET - OFFSET_INCREMENT * depth;
+};
+
+const Hitboxes: React.FC<Props> = ({ zoneId, parentZoneId, enableRootHitboxes }) => {
+  const tree = useTreeStore((state) => state.tree);
+  const isDraggingOnCanvas = useDraggedItemStore((state) => state.isDraggingOnCanvas);
+  const zoneDepth = useMemo(
+    () => getItemDepthFromTree({ id: parentZoneId }, tree),
+    [tree, parentZoneId],
+  );
+  const [fetchDomRect, setFetchDomRect] = useState(Date.now());
+
+  useEffect(() => {
+    /**
+     * A bit hacky but we need to wait a very small amount
+     * of time to fetch the dom getBoundingClientRect once a
+     * drag starts because we need pre-drag styles like padding
+     * applied before we calculate positions of hitboxes
+     */
+    setTimeout(() => {
+      setFetchDomRect(Date.now());
+    }, 50);
+  }, [isDraggingOnCanvas]);
+
+  const hitboxContainer = useMemo(() => {
+    return document.querySelector('[data-ctfl-hitboxes]');
+  }, []);
+
+  const domRect = useMemo(() => {
+    return document.querySelector(`[${CTFL_ZONE_ID}="${zoneId}"]`)?.getBoundingClientRect();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [zoneId, fetchDomRect]);
+
+  const zones = useZoneStore((state) => state.zones);
+
+  const direction = zones[parentZoneId]?.direction || 'vertical';
+  const isVertical = direction === 'vertical';
+
+  const isRoot = parentZoneId === ROOT_ID;
+
+  const getStyles = useCallback(
+    (direction: HitboxDirection): CSSProperties => {
+      if (!domRect) {
+        return {
+          display: 'none',
+        };
+      }
+
+      const { width, height, top, left, bottom, right } = domRect;
+
+      const isDeepZone = zoneDepth > 5;
+      switch (direction) {
+        case HitboxDirection.TOP:
+          return {
+            width,
+            height: HITBOX_HEIGHT,
+            top: top - calcOffsetDepth(zoneDepth),
+            left,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.BOTTOM:
+          return {
+            width,
+            height: HITBOX_HEIGHT,
+            top: bottom + calcOffsetDepth(zoneDepth),
+            left,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.LEFT:
+          return {
+            width: HITBOX_WIDTH,
+            height: height - HITBOX_HEIGHT,
+            left: left - calcOffsetDepth(zoneDepth) - HITBOX_WIDTH / 2,
+            top: top + HITBOX_HEIGHT / 2,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.RIGHT:
+          return {
+            width: HITBOX_WIDTH,
+            height: height - HITBOX_HEIGHT,
+            left: right - calcOffsetDepth(zoneDepth) - HITBOX_WIDTH / 2,
+            top: top + HITBOX_HEIGHT / 2,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.SELF_VERTICAL: {
+          if (height > 60 && !isDeepZone) {
+            return { display: 'none' };
+          }
```

Comment:
Why `60`?

I see the `SELF_HORIZONTAL` case uses `width > DRAGGABLE_WIDTH`. Would it make sense to have this condition use `height > DRAGGABLE_HEIGHT * 2` instead?